server/src/modules/org_environment_variables/is-public.guard.ts

```diff
@@ -0,0 +1,13 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { AppsService } from '@services/apps.service';
+
+@Injectable()
+export class IsPublicGuard implements CanActivate {
+  constructor(private appsService: AppsService) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const { params } = context.switchToHttp().getRequest();
+    const app = await this.appsService.findBySlug(params['app_slug']);
```

Comment:
```
if (!request?.params?.app_slug) {
  return false;
}
const request = context.switchToHttp().getRequest();
const app = await this.appsService.findBySlug(request.params.app_slug);
request.app = app;
return !!app?.isPublic;
```
Shall we check if `app_slug` is present in request and set the app in request if app available, 
and get the `app` from controller using decorator?